### DISCUSSION TITLE
Can ImageMagick convert animated Gif to animated Webp?

### DISCUSSION THREAD

**Discussion by tempmoro** [ImageMagick/ImageMagick#3607] [General]
Hi, all

Can anyone try to convert an animated gif to animated WebP? 

From my research, but with old post, it says not possible, and need to use google tool. 

However, official website of ImageMagick says they support webp, but is not clear whether  it is possible to convert animated gif to webp? 

Could you please advise me whether it's possible with ImageMagick? 

Many thanks in advance,

**Comment by fmw42** [comment]
It seems to work fine for me on Imagemagick 7.0.11.9 Q16 Mac OSX Sierra with webp 1.0.3. I created a gif animation from 3 images. Then converted it. Then viewed it with magick display.

**Reply by tempmoro** [reply]
Thanks for reply, fmw42. 

Can you direct me what command have you used for it, please? I am newbie for Imagemagick.

**Reply by fmw42** [reply]
I simply did

`magick image.gif image.webp`

but note that you have to have libwebp installed with Imagemagick. It should show up in your list of delegates when you do 1magick -version1, if you have installed it. Note that image.gif was an animated gif that I had created from 3 single images just to see that the webp animates.

**Reply by fmw42** [reply]
For novices, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli

**Reply by tempmoro** [reply]
Super thanks, Weinhaus! 

It's very helpful information. I was struggling with the conversion made to a flat image from animated gif. 

The information would solve my issue.

**Reply by fmw42** [reply]
If you still have an issue, let us know and we can try to help.

**Reply by sergeevabc** [reply]
magick frames/*.png out.webp works on my end, but what options could be tuned here?

**Reply by paul-mesnilgrente** [reply]
I ran `convert image.gif image.webp` but it still splits it.

Running `magick -version` gives me: `Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib`. Note there's webp, but not libwebp, is that fine? I just installed imagemagick from the ubuntu 20.04 packages. I have libwebp6 installed as well as a package.

**Comment by myworkfloww** [comment]
Hi, This command working perfectly in terminal---> `convert image.gif image.webp` but I need a shell script code for Automator for converting animated gif to the animated webp with one click. any help, please...

`/usr/local/bin/convert .gif "$@" ~/Desktop/animated.webp`

this command is working but giving an error after conversion of the file.

`The action “Run Shell Script” encountered an error: “convert: unable to open image '.gif': No such file or directory @ error/blob.c/OpenBlob/3533.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/572.”`

**Reply by fmw42** [reply]
What exactly do you mean by you need a shell script for Animator.  The command you use would do what  you want. You just need to build the AppleScript command to call the shell script. See do shell script at https://developer.apple.com/library/archive/technotes/tn2065/_index.html for example. You may have to provide the full path to your convert and preface with bash
